packages/std/src/results/cosmos_msg.rs

```diff
@@ -109,6 +109,20 @@ pub enum WasmMsg {
         /// optional human-readbale label for the contract
         label: Option<String>,
     },
+    /// Migrates a given contracts to use new wasm code. Passes a MigrateMsg to allow us to
+    /// customize behavior.
+    ///
+    /// Only the contract admin (as defined in wasmd), if any, is able to make this call.
+    ///
+    /// This is translated to a [MsgMigrateContract](https://github.com/CosmWasm/wasmd/blob/v0.14.0/x/wasm/internal/types/tx.proto#L86-L96).
+    /// `sender` is automatically filled with the current contract's address.
+    Migrate {
+        contract_addr: HumanAddr,
+        /// the code_id of the new logic to place in the given contract
+        new_code_id: u64,
+        /// msg is the json-encoded MigrateMsg struct that will be passed to the new code
+        msg: Binary,
+    },
```

Comment:
This is by the way a breaking change as the enum is not `#[non_exhaustive]`. If we ever want to add new cases in 1.x, we should make it `#[non_exhaustive]`.

Reply:
You mean `WasmMsg`? (Which I think is pretty final, we are just finally supporting a feature we added in 0.9).

Or `CosmosMsg` in general, which definitely will expand sometime. I guess I could just add this to all the types here

Reply:
This `WasmMsg` just brought up the topic. I mean CosmosMsg and all its sub-messages. It is just too easy to forget one type, or require a v2 of an existing type later on. If we need to make a CosmWasm 2.0, just because we did not write `#[non_exhaustive]` and use fallback cases in the match, it would be pretty annoying. I guess we do not even `match` those as they are decoded in Go directly from JSON.

Reply:
Huh, I made them non_exhaustive in two commits, and assumed some code would break, at such as places like https://github.com/CosmWasm/cosmwasm/blob/448671c8cb31f0852f27cc5775461ba995f288a1/packages/std/src/mock.rs#L301-L314 but tests keep working. Not sure what this does in the end

Reply:
This is a surprise I also hit recently. Rust is too clever: crate-internal you always must mach all cases. Clippy also marks fallback as dead code, when all cases are matched. Crate-external, you always need a fallback case.